### PR TITLE
Address review comments

### PR DESCRIPTION
**Pull request by alvarolobato**

### CODE REVIEW

**Comment by alvarolobato** [comment]
@reviewbybees esp @jtnord @amuniz

**Comment by amuniz** [comment]
LGTM :bee:

File: src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStep.java [hunk 1]
```diff
@@ -146,6 +149,7 @@ public SettingsProvider getDefaultSettingsProvider() {
             return Jenkins.getInstance().getDescriptorByType(Maven.DescriptorImpl.class);
         }
 
+        @Restricted(NoExternalUse.class)
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStep.java]: NIT: in future add a comment as to why for the next person :-)

File: src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepExecution.java [hunk 2]
```diff
@@ -232,15 +233,10 @@ private String obtainMavenExec() throws AbortException, IOException, Interrupted
...
-                    "WARNING: Step running within docker.image() tool installations are not available see https://issues.jenkins-ci.org/browse/JENKINS-36159. You have specified a Maven installation, which will be ignored.");
-            LOGGER.fine("Ignoring Maven Installation parameter: " + mavenName);
-        }
 
-        if (!withContainer) {
-            LOGGER.fine("Maven Installation parameter: " + mavenName);
-            if (!StringUtils.isEmpty(mavenName)) {
+        if (!StringUtils.isEmpty(mavenName)) {
+            if (!withContainer) {
+                LOGGER.fine("Maven Installation parameter: " + mavenName);
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepExecution.java]: String concatination in logging...

File: src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepExecution.java [hunk 3]
```diff
@@ -293,25 +287,31 @@ private String obtainMavenExec() throws AbortException, IOException, Interrupted
...
                 }
             }
         }
 
-        // if at this point mvnExecPath is still null try to use which command to find a maven executable
-        // this approach is only executed if on unix
-        if (mvnExecPath == null && computer.isUnix()) {
+        // if at this point mvnExecPath is still null try to use which/where command to find a maven executable
+        if (mvnExecPath == null) {
             LOGGER.fine("No Maven Installation or MAVEN_HOME found, looking for mvn executable by using which command");
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepExecution.java]: NIT: searching from maven on the path...
(which is not used on windows).

**Comment by jtnord** [comment]
:bee:

**Pull request merged**